bleak/backends/scanner.py

```diff
@@ -183,74 +159,29 @@ def discovered_devices(self) -> List[BLEDevice]:
         """
         raise NotImplementedError()
 
-    async def get_discovered_devices(self) -> List[BLEDevice]:
-        """Gets the devices registered by the BleakScanner.
-
-        .. deprecated:: 0.11.0
-            This method will be removed in a future version of Bleak. Use the
-            :attr:`.discovered_devices` property instead.
-
-        Returns:
-            A list of the devices that the scanner has discovered during the scanning.
 
-        """
-        warn(
-            "This method will be removed in a future version of Bleak. Use the `discovered_devices` property instead.",
-            FutureWarning,
-            stacklevel=2,
-        )
-        return self.discovered_devices
+def get_platform_scanner_backend_type() -> Type[BaseBleakScanner]:
```

Comment:
It would be nice if this was cached with https://docs.python.org/3/library/functools.html#functools.cache (probably needs `lru_cache(maxsize=None)` instead though since < 3.9 is supported

Reply:
not if it's cached

Reply:
This shaves of a few microseconds on my system.

```
python -m timeit -n 10000 -s 'from bleak import BleakScanner' 's = BleakScanner()' 
```

without caching:
```
10000 loops, best of 5: 14.6 usec per loop
```

with caching:
```
10000 loops, best of 5: 10 usec per loop
```

Reply:
Probably doesn't matter in that case 👍

Reply:
ok, leaving out for now